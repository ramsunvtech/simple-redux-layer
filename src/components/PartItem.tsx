import React from 'react';

// `Material` Components.
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

// Stylesheet.
import useStyles from './PartItem.style';

const Part = ({
  name,
  id,
  status,
}: {
  name: string;
  id: number;
  status: string;
}) => {
  const classes = useStyles();

  return (
    <Card className={classes.root}>
      <CardContent>
        <Typography variant="h5" component="h2"  data-testid="name" >
          {name}
        </Typography>
        <Typography className={classes.pos} color="textSecondary">
          ID: {id}
        </Typography>
        <Typography variant="body2" component="p">
          Status: {status}
        </Typography>
      </CardContent>
    </Card>
  );
}

export default Part;
