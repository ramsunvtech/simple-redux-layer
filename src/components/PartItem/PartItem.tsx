import React from 'react';

// `Material` Components.
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

// `Models`.
import Part from '../../models/Part';

// Stylesheet.
import useStyles from './PartItem.style';

const PartItem = ({
  name,
  id,
  status,
}: Part) => {
  const classes = useStyles();

  return (
    <Card className={classes.root}>
      <CardContent>
        <Typography variant="h5" component="h2"  data-testid="name" >
          {name}
        </Typography>
        <Typography className={classes.pos} color="textSecondary">
          ID: {id}
        </Typography>
        <Typography variant="body2" component="p">
          Status: {status}
        </Typography>
      </CardContent>
    </Card>
  );
}

export default PartItem;
