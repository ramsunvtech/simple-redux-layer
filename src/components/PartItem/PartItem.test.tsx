import React from 'react';
import { render } from '@testing-library/react';

// `Test` Component.
import PartItem from './PartItem';

describe('PartItem', () => {
  it('should have part name', () => {
    const itemProps = {
      name: 'first part',
      id: 1592928182403,
      status: true,
    };
    const { getByTestId, asFragment } = render(<PartItem {...itemProps} />);
    const partName = getByTestId('name');
    expect(partName).toHaveTextContent('first part');
    expect(asFragment(<PartItem {...itemProps}/>)).toMatchSnapshot();
  });
});
