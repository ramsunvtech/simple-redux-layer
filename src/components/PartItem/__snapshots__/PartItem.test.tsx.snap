// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartItem should have part name 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5"
        data-testid="name"
      >
        first part
      </h2>
      <p
        class="MuiTypography-root makeStyles-pos-3 MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        ID: 1592928182403
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        Status: 
      </p>
    </div>
  </div>
</DocumentFragment>
`;
